packages/cli/src/docs-mcp-server/index.ts

```diff
@@ -0,0 +1,213 @@
+import { StdioServerTransport } from '@modelcontextprotocol/sdk/server/stdio.js';
+import { z } from 'zod';
+import { fetchDocsList, fetchDocsPost } from './docs';
+
+/**
+ * Starts the Documentation Message Control Protocol (MCP) server for Langbase.
+ *
+ * This function initializes and runs an MCP server that provides tools for
+ * fetching and searching Langbase documentation. The server includes several tools:
+ *
+ * - docs-route-finder: Searches through documentation routes based on user query
+ * - sdk-documentation-fetcher: Fetches detailed SDK documentation
+ * - examples-tool: Fetches code examples and sample implementations
+ * - guide-tool: Fetches detailed guides and tutorials
+ * - api-reference-tool: Fetches API reference documentation
+ *
+ * The server communicates using a standard I/O transport mechanism.
+ *
+ * @returns {Promise<void>} A promise that resolves when the server completes execution
+ *
+ * @throws Will throw an error if connecting to the transport fails
+ * @throws Will throw an error if there is a fatal error in the main execution
+ */
+export async function docsMcpServer() {
+	const server = new McpServer({
+		name: 'langbase-docs-server',
+		version: '0.1.0'
+	});
+
+	server.tool(
+		'docs-route-finder',
+		"Searches through all available documentation routes and returns relevant paths based on the user's query. This tool helps navigate the documentation by finding the most appropriate sections that match the search criteria.",
+		{
+			query: z.string()
+				.describe(`A refined search term extracted from the user's question.
+            For example, if user asks 'How do I create a pipe?', the query would be 'SDK Pipe'.
+            This should be the specific concept or topic to search for in the documentation.
+            Treat keyword add as create if user ask for Eg. 'How do I add memory to pipe?' the query should be 'create memory'`)
+		},
+		async ({ query }) => {
+			const docs = await fetchDocsList();
+			// search through the docs and return the most relevent path based on the query
+			const docLines = docs.split('\n').filter(line => line.trim());
+
+			const getRelevanceScore = (line: string, searchQuery: string) => {
+				const lowerLine = line.toLowerCase();
+				const lowerQuery = searchQuery.toLowerCase();
+				// Higher score for exact matches
+				if (lowerLine.includes(lowerQuery)) {
+					return 3;
+				}
+
+				// Score based on word matches
+				const queryWords = lowerQuery.split(' ');
+				return queryWords.reduce((score, word) => {
+					return score + (lowerLine.includes(word) ? 1 : 0);
+				}, 0);
+			};
+
+			// Score and sort the documentation entries
+			const scoredDocs = docLines
+				.map(line => ({
+					line,
+					score: getRelevanceScore(line, query)
+				}))
+				.sort((a, b) => b.score - a.score)
+				.filter(doc => doc.score > 0)
+				.slice(0, 3); // Get top 3 most relevant results
+
+			if (scoredDocs.length === 0) {
```

Comment:
Let's be explicit here. Instead of `scoredDocs.length === 0`, we can put it in some variable like

```ts
const hasRelevantDocs = scoredDocs.length === 0;
if (!hasRelevantDocs) {
	return {
		content: [
			{
				type: 'text',
				text:
					'No relevant documentation found for the query: ' +
					query
			}
		]
	};
}
```